### ISSUE TITLE
Servers is allowing mid_loc cache group to map edge servers.

### ISSUE THREAD

**Issue by dmohan001c** [apache/trafficcontrol#5484]
## I'm submitting a ...
-  bug report

## Traffic Control components affected ...
-  CDN in a Box
-  Documentation
-  Traffic Ops
-  Traffic Portal

## Current behavior:
Mid_loc CG is able to map on edge servers

## Expected behavior:
Actually, the EDGE_LOC cache group should be mapped to servers of type EDGE, and the MID_LOC cache group should be mapped to servers of type MID.

## Minimal reproduction of the problem with instructions:
https://tp.nightly.cdnlab.comcast.net/#!/servers -  Create Servers with mismatch type.

**Comment by mitchell852** [comment]
Not sure why server.type vs. server.cachegroup.type validation was never introduced/enforced. Maybe because of a scenario like this:

1. Create server of type EDGE that must be in cachegroup of type EDGE_LOC
2. Try to change the server to type MID - can't be done.

So, it's possible this current behavior is "as designed".

**Comment by zrhoffman** [comment]
> Not sure why server.type vs. server.cachegroup.type validation was never introduced/enforced. Maybe because of a scenario like this:
> 
>     1. Create server of type EDGE that must be in cachegroup of type EDGE_LOC
> 
>     2. Try to change the server to type MID - can't be done.
> 
> 
> So, it's possible this current behavior is "as designed".

The Server could be changed to MID type if it is moved to a a Cache Group of type MID_LOC in the same PUT request.

**Comment by mitchell852** [comment]
> The Server could be changed to MID type if it is moved to a a Cache Group of type MID_LOC in the same PUT request.

yep, you beat me to it. that just occurred to me. i edited my comment.

**Comment by mitchell852** [comment]
Also, remember server types are dynamic. You can create them as you please. So you could create a server type of `FOO`. Which cache group types should FOO server types be limited to? TC is very flexible, hence, the absence of many validation like this.
